documentation/jetty/modules/operations-guide/pages/deploy/index.adoc

```diff
@@ -197,137 +351,127 @@ You can use the features of xref:xml/index.adoc[Jetty XML files] to avoid to har
+The same holds for `+*.properties+` files.
 
-The `{ee-current}-feature.properties` file contains:
+The environment-specific Jetty XML files will be applied to the web application _before_ any Jetty context XML file associated with the web application, present in the `$JETTY_HOME/webapps` directory.
 
-.{ee-current}-feature.properties
-[,properties,subs=attributes+]
-----
-jetty.deploy.environmentXml.feature=etc/{ee-current}-feature.xml
-----
+IMPORTANT: The contents of the environment specific Jetty context XML file may only contain references to classes appropriate for that environment.
 
-The `{ee-current}-context.xml` file contains:
+For example, you can have a `$JETTY_HOME/environments/{ee-current}.xml` file with the following content:
 
-.{ee-current}-context.xml
-[,xml,subs="attributes+,+quotes"]
+.{ee-current}.xml
+[,xml,subs=+attributes]
 ----
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://jetty.org/configure_10_0.dtd">
 
-<Configure id="wac" class="org.eclipse.jetty.{ee-current}.webapp.WebAppContext">
+<Configure class="org.eclipse.jetty.{ee-current}.webapp.WebAppContext">
   <Call name="setAttribute">
-    <Arg>common</Arg>
+    <Arg>com.acme.{ee-current}.feature</Arg>
     <Arg>value</Arg>
   </Call>
 </Configure>
 ----
 
-The `{ee-current}-feature.xml` file contains:
+The `$JETTY_HOME/environments/{ee-current}-extra.xml` file may contain:
 
-.{ee-current}-feature.xml
-[,xml,subs="attributes+,+quotes"]
+.{ee-current}-extra.xml
+[,xml,subs=+attributes]
 ----
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://jetty.org/configure_10_0.dtd">
 
-<Configure id="wac" class="org.eclipse.jetty.{ee-current}.webapp.WebAppContext">
+<Configure class="org.eclipse.jetty.{ee-current}.webapp.WebAppContext">
   <Call name="addEventListener">
     <Arg>
-      <New class="com.example.MyListenerFeature"/>
+      <New class="com.acme.MyListenerFeature"/>
     </Arg>
   </Call>
 </Configure>
 ----
 
-
 The directory structure would look like this:
 
-.directory structure
-[,properties,subs=attributes+]
+[,subs=+attributes]
 ----
 $JETTY_BASE
-|- etc
-│  |-{ee-current}-context.xml
-|  |-{ee-current}-feature.xml
-|- webapps
-    |-{ee-current}.properties
-    |-{ee-current}-feature.properties
-    |-wiki.properties
-    |- wiki.xml
+├── environments
+│   ├── {ee-current}.xml
+│   └── {ee-current}-extra.xml
+└── webapps
+    ├── wiki.properties
+    └── wiki.xml
 ----
 
-The contents of the `$JETTY_BASE/etc/{ee-current}-context.xml` then `$JETTY_BASE/etc/{ee-current}-feature.xml` files will be applied to the `wiki` webapp instance _before_ `wiki.xml`, allowing the contents of the latter to override the contents of the former.
+[[jetty-web-xml]]
```

Comment:
I wouldn't document the `jetty-web.xml` files in the section on deployment. It has nothing to do with deployment, it concerns only webapp startup.

Reply:
You need to know the existence of this feature if you want to properly prepare your application for deployment.

Is there a "webapp startup" documentation? If not, here would be the best place for now.